demisto_sdk/commands/find_dependencies/find_dependencies.py

```diff
@@ -967,7 +992,59 @@ def _find_pack_dependencies(pack_id, id_set, verbose_file, exclude_ignored_depen
         return pack_dependencies
 
     @staticmethod
-    def build_dependency_graph(pack_id, id_set, verbose_file, exclude_ignored_dependencies=True):
+    def build_all_dependencies_graph(pack_ids: list,
+                                     id_set: dict,
+                                     verbose_file: VerboseFile,
+                                     exclude_ignored_dependencies: bool = True) -> nx.DiGraph:
+        """
+        Builds all level of dependencies and returns dependency graph for all packs
+
+        Args:
+            pack_ids (list): pack ids, currently pack folder names is in use.
+            id_set (dict): id set json.
+            verbose_file (VerboseFile): path to dependency explanations file.
+            exclude_ignored_dependencies (bool): Determines whether to include unsupported dependencies or not.
+
+        Returns:
+            DiGraph: all dependencies of given packs.
+        """
+        dependency_graph = nx.DiGraph()
+        for pack in pack_ids:
+            dependency_graph.add_node(pack, mandatory_for_packs=[])
+        for pack in dependency_graph.nodes():
+            dependencies = PackDependencies._find_pack_dependencies(
+                pack, id_set, verbose_file=verbose_file, exclude_ignored_dependencies=exclude_ignored_dependencies)
+            for dependency_name, is_mandatory in dependencies:
+                if dependency_name == pack:
+                    continue
+                if dependency_name not in dependency_graph:
+                    dependency_graph.add_node(dependency_name, mandatory_for_packs=[])
+                dependency_graph.add_edge(pack, dependency_name)
+                if is_mandatory:
+                    dependency_graph.nodes()[dependency_name]['mandatory_for_packs'].append(pack)
```

Comment:
why do we need this direction?

Reply:
In the previous way - when you calculate a graph for each pack - you can just write in it's mandatory dependencies a boolean indicator and it's obvious that. it's mandatory for the source node.

Now that we calculate the whole graph in one time - a boolean is not enough because there is no source node in this case, so the solution I had was to implement it with a list of packs that the current node is mandatory for